src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,106 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.InvalidUserDataException;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private User validUser;
+
+    @BeforeAll
+    static void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void createValidUser() {
+        validUser = new User();
+        validUser.setLogin("existingUser");
+        validUser.setPassword("password123");
+        validUser.setAge(20);
+    }
+
+    private User createUserWithInvalidAge() {
+        User user = new User();
+        user.setLogin("newUser");
+        user.setPassword("password123");
+        user.setAge(17);
+        return user;
+    }
+
+    private User createUserWithShortPassword() {
+        User user = new User();
+        user.setLogin("newUser");
+        user.setPassword("passs");
+        user.setAge(20);
+        return user;
+    }
+
+    private User createUserWithShortLogin() {
+        User user = new User();
+        user.setLogin("usrrr");
+        user.setPassword("password123");
+        user.setAge(20);
+        return user;
+    }

```

Comment:
you don't need those methods, just set validUser login too short for tooShortLogin test. You almost don't use your validUser